internal/testutils/feature.go

```diff
@@ -48,13 +48,13 @@ func CheckFeatureTest(t *testing.T, fn func() error) {
 }
 
 func SkipIfNotSupported(tb testing.TB, err error) {
-	ufe, ok := errors.Cause(err).(*internal.UnsupportedFeatureError)
-	if !ok {
-		return
+	for err != nil {
```

Comment:
I think you've re-implemented `errors.As` here, is there a particular reason for that?

My suggestion would be:
```
	var ufe *internal.UnsupportedFeatureError
	if xerrors.As(err, &ufe) {
		checkKernelVersion(tb, ufe)
		tb.Skip(ufe.Error())
	}
```

Reply:
will update this according to your comment.